.github/workflows/ci-cd.yml

```diff
@@ -54,42 +54,80 @@ jobs:
         with:
           fetch-depth: 0
 
+      - if: steps.conditionals_handler.outputs.is_push_to_default_branch == 'true'
+        name: Bump GH tag
+        id: tag_generator
+        uses: mathieudutour/github-tag-action@v5.4
+        with: 
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          default_bump: false
+          release_branches: ${{ steps.data_gatherer.outputs.default_branch }}
+
       - name: Build solution
         id: build_solution
         shell: pwsh
         run: |
-          dotnet build .\GuardClauses.sln --configuration Release
+          $gitNewVersion = if ("${{ steps.tag_generator.outputs.new_version }}") {"${{ steps.tag_generator.outputs.new_version }}"} else {$null}
+          $projectCurrentVersion = ([xml](Get-Content .\src\GuardClauses\GuardClauses.csproj)).Project.PropertyGroup.Version
+          $projectNewVersion = $gitNewVersion ?? $projectCurrentVersion
+          dotnet build .\GuardClauses.sln --configuration Release -property:Version=$projectNewVersion
 
       - name: Run Unit Tests
         id: run_unit_tests
         shell: pwsh
         run: |
           dotnet test .\GuardClauses.sln --verbosity normal
```

Comment:
Consider --no-restore/--no-build flags and a --configuration property. We will discuss this tomorrow.